src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs

```diff
@@ -155,10 +155,334 @@ public SyntaxNode MethodDeclaration(IMethodSymbol method, IEnumerable<SyntaxNode
         }
 
         /// <summary>
-        /// Creates a method declaration.
+        /// Creates an implicit conversion declaration.
+        /// </summary>
+        public virtual SyntaxNode ImplicitConversionDeclaration(
+            IEnumerable<SyntaxNode> parameters = null,
+            SyntaxNode returnType = null,
+            Accessibility accessibility = Accessibility.NotApplicable,
+            DeclarationModifiers modifiers = default(DeclarationModifiers),
+            IEnumerable<SyntaxNode> statements = null)
+        {
+            throw new NotImplementedException();
```

Comment:
We usually don't throw `NotImplementedException`. The [docs say](https://msdn.microsoft.com/en-us/library/system.notimplementedexception.aspx):

> 1. You've designed a type in which you want to include a particular member as part of its public interface, but intend to provide the implementation in a later version. In this case, you can throw a NotImplementedException exception for versions on which the functionality is not implemented, and provide an implementation for later versions.
>
> 2. You've designed a type with one or more members that are available on some platforms or in some versions but not on others. In this case, you can throw a NotImplementedException exception on platforms on which the functionality is not supported.
>
> 3. You're creating an abstract base class with some members for which you want to provide no implementation but expect derived class to provide the implementation when they override the member. In this case, your member can throw a NotImplementedException exception.

I'd say that (1) is extremely odd and you shouldn't be doing this. Until you implement something you don't know whether the design works. (2) is just plain wrong, the code should throw `PlatformNotSupportedException`. (3) seems more reasonable and is basically the case here.

My high-level feedback would be that you should avoid adding virtual members that throw. Instead, I'd try to ensure that design has a reasonable default implementation, for instance, by forwarding to existing APIs. If that's not possible, at least add some actionable message. Simply failing with `NotImplementedException` will look like a bug in Roslyn.